### DISCUSSION TITLE
Hono connection for ditto

### DISCUSSION THREAD

**Discussion by BHUVANESHWARI967** [eclipse-ditto/ditto#1719] [Q&A] [answered]
Hi all,

Is it possible to create a Hono connection for local ditto through kafka (without using c2e package). If there is a way can you provide me with an sample example  and steps to do so and what are the installation required apart from ditto.

**Comment by casafurix** [comment]
Hello @BHUVANESHWARI967 , I think we both are working on similar things, even I need help with the same, are you by any chance trying to create a Digital Twin for an application?

**Comment by dimabarbul** [comment]
Hello.

There is a special [Eclipse Hono binding](https://eclipse.dev/ditto/connectivity-protocol-bindings-hono.html) which was added, I believe, in version 3.2. I didn't try configuring it by myself, but according to description this might be the one you're looking for.

If you use Ditto with version prior to 3.2, I guess, you'll need to configure [Apache Kafka 2.x binding](https://eclipse.dev/ditto/3.1/connectivity-protocol-bindings-kafka2.html).

Hope this helps.

**Comment by BHUVANESHWARI967** [comment]
Hi,

I have tried creating kafka connection with local ditto which was installed by docker-compose but it showed misconfigured. So, I want an proper documentation for establishing the hono binding with ditto. What are the setup required to achieve this.q

**Reply by thjaeckle** [reply]
As @dimabarbul already answered with the correct documentation locations, I have nothing to add.
There is no other secret documentation.

What you maybe should check is to use the correct hostname to connect to Kafka.
A common mistake by Docker newbies is to use `localhost` which surely is not a correct hostname when running in a docker network.

**Reply by BHUVANESHWARI967** [reply]
Forgive me for asking the clarification of simple doubt as I'm new to this technologies, but my question was is it possible to connect hono installed in minikube cluster be used to create a connection with ditto which is installed using docker-compose.

**Reply by thjaeckle** [reply]
> but my question was is it possible to connect hono installed in minikube cluster be used to create a connection with ditto which is installed using docker-compose.

You did not ask that.
I assume that it should be possible but I can't provide documentation for such a strange setup.

**Reply by BHUVANESHWARI967** [reply]
Could you please describe a simpler setup to do this?

**Reply by thjaeckle** [reply, marked as answer]
For a local development setup I would, in favor of simplicity, always choose docker compose for all dependencies.

For a more advanced and productive setup deploy all to Kubernetes.
Ditto supports both and provides for k8s also a Helm chart.
